jib-core/src/main/java/com/google/cloud/tools/jib/image/json/HistoryObjectTemplate.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.image.json;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.google.cloud.tools.jib.json.JsonTemplate;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class HistoryObjectTemplate implements JsonTemplate {
```

Comment:
javadoc

Reply:
Also, was there a reason for having this outside of `ContainerConfigurationTemplate` (besides just a long file)? If we keep it outside, might want to name it like `ContainerConfigurationHistoryObjectTemplate` for something.

Reply:
Wasn't sure if it was a good idea to use nested classes in this case since `HistoryObjectTemplate` is used in multiple places outside of `ContainerConfigurationTemplate`. But I could nest it if you think it's a better idea.

Reply:
It's mostly that even though it is used outside of `ContainerConfigurationTemplate`, it is just a type of object inside the config JSON, so some sort of association there that ties them closely would probably make more sense for the class relationship.